### PR TITLE
Polish the circuit breaker code.

### PR DESCRIPTION
**Pull request by louyuting**
### Describe what this PR does / why we need it
Refine the way to instance circuit breaker rule.

1. Using unified function `circuitbreaker.NewRule` to instance circuit rule instead of three function:
`circuitbreaker.NewSlowRtRule、circuitbreaker.NewErrorRatioRule、circuitbreaker.NewErrorCountRule`
2. Using options to set rule‘s field and Provide default field value for circuit breaker rule.

### Special notes for reviews
Circuit breaker wiki had been updated, please review.

[https://github.com/alibaba/sentinel-golang/wiki/熔断降级](https://github.com/alibaba/sentinel-golang/wiki/熔断降级)

### CODE REVIEW

**Comment by louyuting** [comment]
Please let me know if any codes/comments/logs need to polish in circuit breaker components.

@sczyh30

File: core/circuitbreaker/stat_slot.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 )
 
 // MetricStatSlot records metrics for circuit breaker on invocation completed.
+// MetricStatSlot must be filled into slot chain is circuit breaker is alive.
```
Comment (sczyh30) [inline core/circuitbreaker/stat_slot.go:8]: What does this line mean?

File: ext/datasource/helper.go [hunk 2]
```diff
@@ -142,48 +142,55 @@ func CircuitBreakerRulesJsonConverter(src []byte) (interface{}, error) {
...
-	var rules []circuitbreaker.Rule
-	if val, ok := data.([]circuitbreaker.Rule); ok {
+	var rules []cb.Rule
+	if val, ok := data.([]cb.Rule); ok {
 		rules = val
 	} else {
 		return Error{
 			code: UpdatePropertyError,
-			desc: fmt.Sprintf("Fail to type assert data to []circuitbreaker.Rule, in fact, data: %+v", data),
+			desc: fmt.Sprintf("Fail to type assert data to []cb.Rule, in fact, data: %+v", data),
```
Comment (sczyh30) [inline ext/datasource/helper.go:190]: Maybe no need to change the pkg name in the log?

**Comment by sczyh30** [comment]
Thanks!

**Pull request merged**